src/Temporalio/Worker/IWorkerClient.cs

```diff
@@ -0,0 +1,20 @@
+namespace Temporalio.Worker
+{
+    /// <summary>
+    /// Client that can be used to create a worker.
+    /// </summary>
+    public interface IWorkerClient
```

Comment:
This interface is implemented by `TemporalClient` and is technically all that is needed for a worker to function. This is a nice abstraction separation (and the reason `IBridgeClientProvider` exists). This means users can have their own impl of `ITemporalClient` (e.g. a wrapper around ours) and since this interface is still required to be impl'd, it'll work for a worker.